src/foam/java/refinements.js

```diff
@@ -35,7 +35,8 @@ foam.LIB({
           return b ? "true" : "false";
         },
         Number: function(n) {
-          return '' + n + (n > Math.pow(2, 31) ? 'L' : '');
+          return '' + n +
+            (n > Math.pow(2, 31) || n < -Math.pow(2,31) ? 'L' : '');
```

Comment:
You can use `Math.abs` to make this a bit shorter
`return '' + n + (Math.abs(n) > Math.pow(2, 31) ? 'L' : '');`